demo/vl-map-draw-actions.html

```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+    <title>vl-map-draw-action voorbeeld</title>
+    <script type="module" src="/node_modules/vl-ui-demo/dist/vl-demo-all.js"></script>
+    <script type="module" src="/node_modules/vl-ui-body/dist/vl-body.js"></script>
+
+    <script type="module" src="/src/vl-map-all.js"></script>
+
+    <link rel="stylesheet" type="text/css" href="/node_modules/vl-ui-body/dist/style.css" />
+</head>
+
+<body is="vl-body">
+    <vl-demo-page data-vl-webcomponent="vl-map-draw-action">
+        <vl-demo data-vl-title="Kaart met punt teken actie">
+            <vl-map id="map-with-draw-point-action">
+                <vl-map-baselayer-grb-gray></vl-map-baselayer-grb-gray>
+                <vl-map-baselayer-grb></vl-map-baselayer-grb>
+                <vl-map-baselayer-grb-ortho></vl-map-baselayer-grb-ortho>
+                <vl-map-layer id="point-layer">
+                    <vl-map-layer-circle-style></vl-map-layer-circle-style>
+                    <vl-map-draw-point-action id="draw-point-action" data-vl-default-active></vl-map-draw-point-action>
```

Comment:
ik zou een actie niet onder een map steken, maar via de title van de layer of zo de beide aan mekaar koppelen.  een actie is niet "onlosmakelijk" verbonden met 1 laag.

Reply:
Inderdaad, het is momenteel mogelijk om een actie op map niveau te declareren en programmatisch de koppeling te voorzien van de layer via een setter. Bij de vl-map-select-action wordt dit getoond in de demo:

```HTML
<vl-map id="map-with-select-action">
    <vl-map-baselayer-grb-gray></vl-map-baselayer-grb-gray>
    <vl-map-baselayer-grb></vl-map-baselayer-grb>
    <vl-map-baselayer-grb-ortho></vl-map-baselayer-grb-ortho>
    <vl-map-layer id="map-layer" data-vl-name="laag" data-vl-features='
        {
            "type": "FeatureCollection",
            "features": [{
                "type": "Feature",
                "id": 1,
                "geometry": {
                    "type": "Point",
                    "coordinates": [147055.0,197908.0]
                }
            }, {
                "type": "Feature",
                "id": 2,
                "geometry": {
                    "type": "Point",
                    "coordinates": [149055.0,199908.0]
                }
            }, {
                "type": "Feature",
                "id": 3,
                "geometry": {
                    "type": "Point",
                    "coordinates": [151055.0,201908.0]
                }
            }]
        }'>
        <vl-map-layer-circle-style></vl-map-layer-circle-style>
    </vl-map-layer>
    <vl-map-select-action id="select-action" data-vl-default-active>
        <vl-map-layer-circle-style data-vl-text-color="#000" data-vl-color="#FFE615" data-vl-border-color="#FFE615"></vl-map-layer-circle-style>
    </vl-map-select-action>
</vl-map>
```

De optie, vereenvoudiging, bestaat om een action binnen een layer te declareren zodat de koppeling automatisch gebeurd maar is niet verplicht om te gebruiken. Programmatische koppeling via setter is dus mogelijk. Indien nodig kan ook nog een HTML layer attribuut koppeling voorzien worden.

Reply:
Ja, maar ge gaat ook andere acties hebben die ge aan meer dan 1 laag wilt koppelen, en aangezien het de bedoeling was om 'zo declaratief mogelijk' te gaan, zou ik zeker ook nog voorzien om 'non-Javascript' te kunnen koppelen bij die use case ...

Reply:
Ik heb een uitbreiding voorzien met attribuut koppeling.

Reply:
voor mij hoeft het feit dat een actie binnen een layer kan staan dan niet meer. maar daar ga ik niet heel moeilijk over doen als je dat liever laat staan, al is het natuurlijk wel een manier extra die je dan moet blijven ondersteunen ...

Reply:
Ik ben overtuigd dat een kaartactie vaak maar verbonden zal zijn met één kaartlaag en dan is deze vereenvoudiging voor afnemers zeer handig, net zoals een lege vl-map-layer-switcher dat kan zijn. De extra code die hiervoor moet voorzien en onderhouden worden is minimaal.

Reply:
zodra er "een" actie is die met meerdere kaartlagen kan werken, is het misschien wel dus geen geweldig idee om "layer" en niet "layerS" in de base class te voorzien. 
het feit dat ge binnen een layer een actie definieert staat daar op zich dan misschien wel los van, maar dan moet ge de logica om naar uw "ene" laag te zoeken misschien wel aanpassen.

Reply:
Tenzij op dat moment bijvoorbeeld een LayersMapAction geïntroduceerd wordt natuurlijk.

Reply:
dat is zeker een optie, het in den "base" laten steken zou ik alleen niet doen ...